stream_consumer.go

```diff
@@ -31,10 +28,54 @@ type StreamEndpointConfig struct {
 	backoffMaxDelay time.Duration
 }
 
-type Consumer struct {
-	StreamName string
-	EvtChan    chan *stream.Event
+type StreamConsumer interface {
```

Comment:
Are you making an interface, so it's easier to mock, or replace gRPC?

Reply:
I created it to have a common interface between consumer (old struct, still usable) and registeredConsumer (new struct)